packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js

```diff
@@ -1,621 +1,944 @@
+
+          await finder.changeImplementationAddress(
+            utf8ToHex(interfaceName.IdentifierWhitelist),
+            identifierWhitelist.address
+          );
+
+          mockOracle = await MockOracle.new(finder.address, timer.address);
+          await finder.changeImplementationAddress(utf8ToHex(interfaceName.Oracle), mockOracle.address);
+        });
 
-        // Calling create from a new address will create a new position and this should be added the the client.
-        await emp.create(
-          { rawValue: convertCollateral("100") },
-          { rawValue: convertSynthetic("45") },
-          { from: sponsor2 }
-        );
-        await updateAndVerify(
-          client,
-          [sponsor1, sponsor2],
-          [
-            {
-              sponsor: sponsor1,
-              numTokens: convertSynthetic("100"),
-              amountCollateral: convertCollateral("20"),
-              hasPendingWithdrawal: false,
-              withdrawalRequestPassTimestamp: "0",
-              withdrawalRequestAmount: "0"
-            },
+        beforeEach(async function() {
+          await timer.setCurrentTime(startTime - 1);
+          constructorParams = await _createConstructorParamsForContractVersion(currentVersionTested, currentTypeTested);
+
+          emp = await financialContractInstance.new(constructorParams);
+          await syntheticToken.addMinter(emp.address);
+          await syntheticToken.addBurner(emp.address);
+
+          await collateralToken.approve(emp.address, convertCollateral("1000000"), { from: sponsor1 });
+          await collateralToken.approve(emp.address, convertCollateral("1000000"), { from: sponsor2 });
+          await syntheticToken.approve(emp.address, convertSynthetic("100000000"), { from: sponsor1 });
+          await syntheticToken.approve(emp.address, convertSynthetic("100000000"), { from: sponsor2 });
+
+          // If we are testing a perpetual then we need to apply the initial funding rate to start the timer.
+          await emp.setCurrentTime(startTime);
+          if (currentTypeTested == "Perpetual") await emp.applyFundingRate();
+
+          // The ExpiringMultiPartyClient does not emit any info `level` events.  Therefore no need to test Winston outputs.
+          // DummyLogger will not print anything to console as only capture `info` level events.
+          dummyLogger = winston.createLogger({
+            level: "info",
+            transports: [new winston.transports.Console()]
+          });
+
+          client = new ExpiringMultiPartyClient(
+            dummyLogger,
+            financialContractInstance.abi,
+            web3,
+            emp.address,
+            testConfig.collateralDecimals,
+            testConfig.syntheticDecimals,
+            testConfig.priceFeedDecimals,
+            currentTypeTested // either ExpiringMultiParty OR Perpetual depending on the test
+          );
+        });
+        versionedIt(["any"])("Returns all positions", async function() {
```

Comment:
here you can see how the new test structure is used. This test is meant to be run on `any` contract version (meaning all contract versions specified should be tested).

Reply:
This is a pretty cool feature! Does this mean that the test suite takes 5x as long to run now?

Reply:
it takes 3x as long as right now we running on three versions. I'm working (in a separate PR) on speeding up CI so hopefully, it's not too bad to the overall run time!